src/storage/src/store.rs

```diff
@@ -394,6 +394,18 @@ impl From<TracedWriteOptions> for WriteOptions {
     }
 }
 
+pub trait CheckOldValueEquality = Fn(&Bytes, &Bytes) -> bool + Send + Sync;
+
+pub static CHECK_BYTES_EQUAL: LazyLock<Arc<dyn CheckOldValueEquality>> =
+    LazyLock::new(|| Arc::new(|first: &Bytes, second: &Bytes| first == second));
+
+#[derive(Default, Clone)]
+pub enum OpConsistencyLevel {
+    #[default]
+    Inconsistent,
+    ConsistentOldValue(Arc<dyn CheckOldValueEquality>),
+}
```

Comment:
It appears that we only have two kinds of `impl Check` now: equal by bytes, or by values. Is it possible to make it an `enum` to get rid of `dyn` here?

Reply:
Since the performance is not a major concern, I prefer keeping it simple.